### ISSUE TITLE
Missing parenthesis in Enum methods throws error in platform but not in compilation time

### ISSUE THREAD

**Issue by sergiogp1** [microsoft/AL#7629]
**1. Describe the bug**
Missing parenthesis in Enum methods throws error in platform but not in compilation time. 

**2. To Reproduce**

1. Create an Enum

``` AL
enum 70021 "ABC Sale Type"
{
    value(0; A) {}
    value(1; B) {}
}
```
2. Add this code
``` AL
Txt := Enum::"ABC Sale Type".Names.Get(Enum::"ABC Sale Type".Ordinals.IndexOf(1))
```

**3. Expected behavior**
AL compiler should throw a clear error because error is thrown later in platform.

**4. Actual behavior**
You can compile but an error is thrown when you try to publish: _Unexpected value 'NoConversion' of type 'Microsoft.Dynamics.Nav.CodeAnalysis.ConversionKind') Unexpected value 'NoConversion' of type 'Microsoft.Dynamics.Nav.CodeAnalysis.ConversionKind'_

**5. Solution:**
Add parenthesis to Names() method and Ordinals() methods
``` AL
Txt := Enum::"ABC Sale Type".Names().Get(Enum::"ABC Sale Type".Ordinals().IndexOf(1))
```

**6. Versions:**

- AL Language: I guess 12.1
- Visual Studio Code: 1.85.1
- Business Central: 23.1

### Final Checklist

Please remember to do the following:

* [X] Search the issue repository to ensure you are reporting a new issue
* [X] Reproduce the issue after disabling all extensions except the AL Language extension
* [X] Simplify your code around the issue to better isolate the problem

**Comment by BazookaMusic** [comment]
Thank you for the repro, it's super useful. Happy new year!

**Comment by bjarkihall** [comment]
Duplicate/related to #7477 ?

**Comment by thloke** [comment]
This specific bug should be fixed (#7477 is still open though)
